### PR TITLE
feat: add all command modules

### PR DESCRIPTION
**Pull request by needhamgary**
Also made plugins optional on Executable classes.

### CODE REVIEW

**Comment by needhamgary** [comment]
I attempted to run eslint format, but it was denied due to `//@ts-ignore`'s in the Executable Classes.

File: src/core/modules.ts [hunk 1]
```diff
@@ -1,5 +1,15 @@
-import { ClientEvents } from 'discord.js';
-import { CommandType, EventType, PluginType } from '../core/structures';
+import {
```
Comment (jacoobes) [inline src/core/modules.ts:1]: import type { 
// all your types
} from 'discord.js'

File: src/core/modules.ts [hunk 2]
```diff
@@ -30,6 +41,185 @@ export function commandModule(mod: InputCommand): CommandModule {
...
+    execute: (ctx: RoleSelectMenuInteraction) => Awaitable<unknown>;
+}) {
+    return commandModule({
+        type: CommandType.RoleSelect,
+        ...mod,
+    });
+}
+
+/**
+ * @since 3.2.0 The wrapper function to create Msg Context2 Menu Commands for sern.
```
Comment (jacoobes) [inline src/core/modules.ts:208]: typo! 
Msg Context2 -> Msg Context

Comment (jacoobes) [inline src/core/modules.ts:208]: typo! 
Msg Context2 -> Msg Context

File: src/core/modules.ts [hunk 3]
```diff
@@ -1,5 +1,15 @@
-import { ClientEvents } from 'discord.js';
```
Comment (jacoobes) [inline src/core/modules.ts:1]: merge this import with other import from 'discord'

Comment (jacoobes) [inline src/core/modules.ts:1]: merge this import with other import from 'discord'

**Review by jacoobes — CHANGES_REQUESTED** [review]

**Comment by jacoobes** [comment]
a little old, and i could figure out the correct typings for dis

**Pull request closed without merging**